Improve navigation

max-elia opened the pull request:
* Make the top Stepper clickable
*Simplify 'completed' logic (mark all steps below the actual step as 'completed')
*Provide option to go back to overview
*Remove Finish Step
Closes #21

Signed-off-by: Max Elia Schweigkofler <max_elia@hotmail.de>

eneufeld commented on react-app/src/components/StepperComponent.tsx:
```diff
@@ -94,11 +94,13 @@ const getSteps = (tutorialExercises: Exercise[] | undefined) => {
 const StepperComponent = (props: StepperComponentProps) => {
   const classes = useStyles();
   const [activeStep, setActiveStep] = useState(props.startStep);
+  const [completed, setCompleted] = React.useState(new Set<number>());
```
why use a set and not a simple array?

max-elia replied:
I simplified the 'completed' logic and got rid of the property altogether.

eneufeld reviewed (approved):
Looks good overall, thank you

max-elia committed:
Improve navigation

* Make the top Stepper clickable
*Provide option to go back to overview
*Remove Finish Step
*Simplify 'completed' logic (mark all steps below the actual step as 'completed')
Closes #21

Signed-off-by: Max Elia Schweigkofler <max_elia@hotmail.de>
